### ISSUE TITLE
Add CIBA support

### ISSUE THREAD

**Issue by CrazyParanoid** [spring-projects/spring-security#14725]
Need to add [https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#rfc.section.7](CIBA) support.

**Comment by kpur-sbab** [comment]
We are migrating our custom build auth solution to Spring authorization server then realized the need of CIBA support.
We basically in need of integrating to Swedish BankID https://developers.bankid.com/api-references/auth--sign/overview
where the user authenticates by scanning the QR code in his/her own mobile device and the auth result is actually obtained by polling to an endpoint offered by the Swedish BankID.
Basically these are the steps involved,
1. Initiate authentication on the users device with /auth endpoint
2. Poll for the result in /collect
3. /cancel to cancel the autentication
The API is pretty straight forward and CIBA looks a perfect fit for this use case.

Kindly correct me if am wrong

**Comment by franticticktick** [comment]
Hey @jzheaux , can we think about the design of this feature? :)

**Comment by jgrandja** [comment]
@franticticktick @kpur-sbab Please see [spring-authorization-server#1938](https://github.com/spring-projects/spring-authorization-server/issues/1938#issuecomment-2762199426)

**Comment by kpur-sbab** [comment]
@jgrandja Thanks for the reply.
I agree there are several flavors that needs a implementation but our team will reevaluate if we can use [this](https://docs.spring.io/spring-authorization-server/reference/guides/how-to-ext-grant-type.html) to implement CIBA support in Spring auth server.
We would appreciate if you could let us know if this feature in spring auth server is totally flexible since I cannot see more documentation on this creation of custom grant type

**Comment by jgrandja** [comment]
@kpur-sbab 

> our team will reevaluate if we can use [this](https://docs.spring.io/spring-authorization-server/reference/guides/how-to-ext-grant-type.html) to implement CIBA support in Spring auth server

I haven't read the CIBA spec in full but at a quick glance it does seem involved and would require a **new Backchannel Authentication Endpoint** and some enhancements to the [Token Endpoint](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#rfc.section.10.1).

**Comment by kpur-sbab** [comment]
@jgrandja 

> I haven't read the CIBA spec in full but at a quick glance it does seem involved 

Do you think still spring auth server will still down prioritize it or it can be considered for enhancement?
